### PR TITLE
chore: Update project check-in and goal update forms

### PR DESCRIPTION
**Pull request by Rockyy174**
I've updated the forms used to create and edit project check-ins and goal updates.

This PR also includes a migration which adds `status` to the goal updates schema. In the next PRs, I intend to update the goal update forms, so that the update status is also saved.

### CODE REVIEW

File: assets/js/pages/ProjectCheckInEditPage/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+function DescriptionSection({ checkIn }: { checkIn: ProjectCheckIn }) {
+  assertPresent(checkIn.project, "project must be present in checkIn");
+
+  const mentionSearchScope = { type: "project", id: checkIn.project.id! } as const;
+
+  return (
+    <div>
+      <div className="text-lg font-bold mb-2">2. What's new since the last check-in?</div>
+
+      <Forms.RichTextArea
```
Comment (shiroyasha) [inline assets/js/pages/ProjectCheckInEditPage/Form.tsx:88]: Can we use the label for this? (or is it visually not the same?)

File: assets/js/components/Forms/SelectStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import * as Popover from "@radix-ui/react-popover";
+import * as People from "@/models/people";
+
+import { Status, Placeholder, StatusOptions } from "@/components/status";
+import { useFieldError, useFieldValue } from "./FormContext";
+
+interface Props {
+  field: string;
+  options: StatusOptions[];
+  reviewer?: People.Person;
```
Comment (shiroyasha) [inline assets/js/components/Forms/SelectStatus.tsx:16]: We should allow setting the label as well.

**Comment by shiroyasha** [comment]
Nice 👏 This was a huge effort and a very good step forward to simplify our codebase.

File: assets/js/pages/ProjectCheckInEditPage/Form.tsx (hunk 1 above)
Comment (Rockyy174) [reply assets/js/pages/ProjectCheckInEditPage/Form.tsx:88]: It looks almost identical. The only difference is the `mb-2` class which adds a space between the label and the element.

Here is what it's looks like without using the label (and how it currently is in production):
tmp-1

Here is with the label:
tmp-2

The same apply for project check-ins and goal updates forms. Do you think this is space between the elements is necessary?

Comment (shiroyasha) [reply assets/js/pages/ProjectCheckInEditPage/Form.tsx:88]: They look almost identical. Lets go with the one that is the default in forms to boost universality in the app.

**Pull request merged**